### PR TITLE
Entity geometry heightReference warnings

### LINKED ISSUE #6976: Polygon error when setting height reference to clamp when perPositionHeight is true
When a polygon entity's `heightReference` is set to `CLAMP_TO_GROUND`, it errors is `perPositionHeight` is enabled, but does not error when disabled.

Brought up in this [forum thread](https://groups.google.com/forum/#!searchin/cesium-dev/jessica$20gettings%7Csort:date/cesium-dev/nPBzPKQhqqg/QnKZgvsXAgAJ). 

[Sandcastle example](https://www.google.com/url?q=https%3A%2F%2Fcesiumjs.org%2FCesium%2FBuild%2FApps%2FSandcastle%2F%23c%3DnVZtc%2BI2EP4rGr4Epq6Aa8jwmmlKuIYpFxhCrtMpHUbYi62JkFxJJqU3%2FPeu38A4OXqt84FYu8%2BjZ1e7K%2B%2BYJi4YHm0XoDXjcqbVjnugyYAMk3XqamAWflVaeJlPtdZbyh0iQQgeGsW9t2AJrznB6CteCU1KtOPwWoZ9TtaqV6m8oZIWsaCvHPJlKQk%2Ba2ZgwvagZ9x9QXSXbJgw4KRWW5LUfT%2FOpTykMup14kFoAwQaS5iPPvibsRBuiIY%2FI67BI1aRLXsBYgMgoRJ7X0lDXMG2YWzKEAlhTAHMI2pDTKBeufQRpFXkB4RbstFqSyKJInxckx6mIgmZGhckUF%2BoNdBE0wIl3aWKMvmYKqsjSJSnKRRK%2BtxGHqDphrZbjZtmq9PGv0brQzM7L8Fs7nLdou1O6%2Fqm1e50Otft9k3m4oGwDO0N2mg00%2FPJRWkA%2BRQyF0Y7kPaBSU%2FE62DHMozsnWu5ktU0%2BZtIpq%2B1%2FLDi5%2FRfxok03HIwdMdEhD8bpUfMDaoFzyNT4rtHvoIxoULRQVw6qQPNToQGwP3AzmEDGqQbZ6qIesuBlfege1%2F34RtSXVaWFfJdst1gQLK3rFwfzvejj9PH0Vn0%2BeNiuSgBVCgfCTF9NqkLlZbQslLrvcUk4k4tWd5rOLn7NFstpquf59Pnx%2FteSfoBG9XAf9IilYT%2FJSUOu4Q6XEp9UcGVVb4vYgXp4WHLZbRdcoWZTrauXTzJy0WQDhg85hPokPMdsrmRxfVUKvfFPgQ6GX1crO6nzz9NRqvhZDz8JR0deQm9xpNjcOrE3skE7NyE0WCHNc%2FgRkU4fgbHLi3ApdJnphM892CnIxkyjTOMM%2FkDjUfMPfgYiqnG6px0k1oBuP5GYCKhCHT%2FFRgH%2FQ7Q%2B0bgUeoJGirD42lgkOJ35pC1Q1yHeH%2Fk7KWxwjyvWqj6rCrwKii1Qoh3SMac1nM3ma7OuVfAQTPtBnskOAop%2BWzxqtScCXTJQ1RC6SS6OU5Mta1%2BYSIMGDo0D7XyDqVy7V7qskINZxVcy6%2BDNA0u26Jg7Cz1cmerlzN%2BLEznWGMOaTUaNad4JT%2FgZYbtOePWDTAaH3LWT8wGdDZ2yPfnC6vp59F89XH6PHfIhwbSJV8O78hbaCYNjv9tgVHzv67p%2BH70uBgvfqv1Kk6lb%2BxewG0a7o98G2JpkUiLKqV1C9sQpYOpryP8HLDUNSberV%2FPQX2P7wj3BstK6ZMCxzjOXmPQsomEeOJ%2F4%2By77dfR%2FwwmVBL%2BdAdasH3sEjRvJ%2BkipbRfx9e3KKuUWDNdYPwH&sa=D&sntz=1&usg=AFQjCNEMXwnjZKEDQABDX8UxnkiBvaOx8Q)

### PR DESCRIPTION
**Pull request by hpinkos**
For #6976

- `oneTimeWarning` if `heightReference` is defined but `height` is not
- `oneTimeWarning` if `extrudedHeightReference` is defined but `extrudedHeight` is not
- `oneTimeWarning` if you try to use `heightReference` with `perPositionHeight` polygons
- `oneTimeWarning` if you try to use `height` and `perPositionHeight` (this used to throw a developer error but doesn't now that we're using `CoplanarPolygonGeometry` for per position height)
- Fix crash if you set `heightReference` and `perPositionHeight`.  Ignores `heightReference` instead.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @hpinkos!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/height-reference-warnings/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by hpinkos** [comment]
Can someone review?  @ggetz or @mramato ?

**Comment by hpinkos** [comment]
I'd like to get this in for the release if possible.  Does anyone have time to review?

**Comment by hpinkos** [comment]
@mramato can you take a look at this please?

**Comment by mramato** [comment]
Taking a look now

File: Source/DataSources/GroundGeometryUpdater.js [hunk 1]
```diff
@@ -120,21 +132,32 @@ define([
     /**
      * @private
      */
-    GroundGeometryUpdater.getGeometryHeight = function(heightProperty, heightReferenceProperty, time) {
-        var heightReference = Property.getValueOrDefault(heightReferenceProperty, time, HeightReference.NONE);
+    GroundGeometryUpdater.getGeometryHeight = function(height, heightReference) {
+        if (!defined(height)) {
+            if (heightReference !== HeightReference.NONE) {
```
Comment (mramato) [inline Source/DataSources/GroundGeometryUpdater.js]: Can't `heightReference` also be `undefined`?  If not, then maybe add a `Check` call for it behind a debug pragma?  I know this is a private function, but it looks like you've moved the responsibility to defaulting `HeightReference.NONE` up to the caller.

**Comment by mramato** [comment]
I think it's just the one comment.  I also ran through Geometry related examples to make sure we weren't triggering any of the warnings ourselves.  Thanks @hpinkos

File: Source/DataSources/GroundGeometryUpdater.js (hunk 1 above)
Comment (hpinkos) [reply Source/DataSources/GroundGeometryUpdater.js]: It should be defaultValued to `NONE` before it gets here, so yeah I'll add a `Check`

**Comment by hpinkos** [comment]
@mramato anything else here?

**Pull request merged**